### PR TITLE
[Mailbox] Make MailboxEventLoop work in ThreadsafePriorityQueue mode

### PR DESCRIPTION
**Pull request by czkkkkkk**

### CODE REVIEW

File: test/threadsafe_priority_queue_unittest.cc [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ protected:
+  void SetUp() {}
+  void TearDown() {}
+};
+
+TEST_F(TestEventQueue, InitDestroy) {
+  ThreadsafePriorityQueue<std::shared_ptr<MailboxEventBase>> queue;
+}
+
+TEST_F(TestEventQueue, Functional) {
```
Comment (kygx-legend) [inline test/threadsafe_priority_queue_unittest.cc:47]: Looks like it's lack of multiple threads case.

**Review by kygx-legend — COMMENTED** [review]
It's good to have such feature but better to give a test about the performance improvement. Even better if that is optional from my perspective.

File: test/threadsafe_priority_queue_unittest.cc (hunk 1 above)
Comment (czkkkkkk) [reply test/threadsafe_priority_queue_unittest.cc:47]: Yes...But it seems that it is hard to verify the result in multiple threads case if we treat it like a black box. Do you have any suggestion? Or test it in h4?

**Comment by czkkkkkk** [comment]
I think we can only test the performance in h4(will test it in h4). The purpose of this changing is to allocate the network to each job evenly according to the size of `BinStream` they send(now we only have job id priority...). So, for h4, I think we don't need previous mode anymore...

File: test/threadsafe_priority_queue_unittest.cc (hunk 1 above)
Comment (kygx-legend) [reply test/threadsafe_priority_queue_unittest.cc:47]: Test case is relative to test coverage. If this is hard to come up with now, we can implement in the future. Not an urgent issue.

**Pull request merged**